### PR TITLE
Modify meta doc

### PR DESCRIPTION
**Pull request by heiglandreas**
This PR contains some changes that should make it more clear why certain decissions were taken the way we took them.

### CODE REVIEW

File: proposed/clock-meta.md [hunk 1]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
...
+* [`ergebnis/clock`](https://packagist.org/packages/ergebnis/clock)
+* [`mangoweb/clock`](https://packagist.org/packages/mangoweb/clock)
+
+(This list is not exhaustive!)
+
+Some providing
 interfaces, and some relying on overloading (extending) the Clock class to mock the
 current time.
 
+These implementations all provide a `now`-method which returns a `DateTimeImmutable`-object. As the `DateTimeImmutable`-object allows to retieve the Unix-Timestamp by calling `getTimestamp()` or `format('u.U')` this interface does not define any special methods to retieve a Unix-Timestamp or any other form of time-information apart from retrieving the `DateTimeImmutable`-object.
```
Comment (shadowhand) [inline proposed/clock-meta.md:61]:
```suggestion
These implementations all provide a `now()` method which returns a `DateTimeImmutable` object. As the `DateTimeImmutable` object allows retrieving the Unix timestamp, by calling `getTimestamp()` or `format('u.U')`, this interface does not define any special methods to retrieve a Unix timestamp or any other time information that is not available from a `DateTimeImmutable` object. 
```

File: proposed/clock-meta.md [hunk 2]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
...
+
+Time by now is defined by interaction of electromagnetic radiation with the excited states of certain atoms where the SI defines one second as the duration of 9192631770 cycles of radiation corresponding to the transition between two energy levels of the ground state of the caesium-133 atom at 0K. This means that retrieving the current time will always return the same time, no matter where it is observed. While the timezone defines *where* the time was observed it does not modify the actual "slice" of time.
+
+This means that for the sake of this PSR the timezone is irrelevant as it is an implementation detail of the code implementing the interface. 
+
+As there is no way to assert a certain timezone in the returnd `DateTimeImmutable` it is up to the implementation to make sure that the timezone is handled according to the Business-Logic. That is either by making sure that a call to `now()` will only return a `DateTimeImmutable`-object with a known timezone (implicit contract) or by making sure that the Business-logic handles the timezone explicitly. This an be done by calling `setTimezone` or by using `getTimestamp` which will automatically return the UTC based Unix-Timestamp.
```
Comment (shadowhand) [inline proposed/clock-meta.md:69]:
```suggestion
It is up to the implementation to make sure that the timezone is handled according to the business logic of the application. That is either by making sure that a call to `now()` will only return a `DateTimeImmutable` object with a known timezone (implicit contract) or by explicitly changing the timezone to be correct for the application. This an be done by calling `setTimezone()` to create a new `DateTimeImmutable` object with the given timezone. 
```

File: proposed/clock-meta.md [hunk 3]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
...
 interfaces, and some relying on overloading (extending) the Clock class to mock the
 current time.
 
+These implementations all provide a `now`-method which returns a `DateTimeImmutable`-object. As the `DateTimeImmutable`-object allows to retieve the Unix-Timestamp by calling `getTimestamp()` or `format('u.U')` this interface does not define any special methods to retieve a Unix-Timestamp or any other form of time-information apart from retrieving the `DateTimeImmutable`-object. 
+
+### 4.2 Timezones
+
+Time by now is defined by interaction of electromagnetic radiation with the excited states of certain atoms where the SI defines one second as the duration of 9192631770 cycles of radiation corresponding to the transition between two energy levels of the ground state of the caesium-133 atom at 0K. This means that retrieving the current time will always return the same time, no matter where it is observed. While the timezone defines *where* the time was observed it does not modify the actual "slice" of time.
+
+This means that for the sake of this PSR the timezone is irrelevant as it is an implementation detail of the code implementing the interface.
```
Comment (shadowhand) [inline proposed/clock-meta.md:67]:
```suggestion
This means that for the sake of this PSR the timezone is considered an implementation detail of the interface. 
```

File: proposed/clock-meta.md [hunk 4]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
...
+
+This means that for the sake of this PSR the timezone is irrelevant as it is an implementation detail of the code implementing the interface. 
+
+As there is no way to assert a certain timezone in the returnd `DateTimeImmutable` it is up to the implementation to make sure that the timezone is handled according to the Business-Logic. That is either by making sure that a call to `now()` will only return a `DateTimeImmutable`-object with a known timezone (implicit contract) or by making sure that the Business-logic handles the timezone explicitly. This an be done by calling `setTimezone` or by using `getTimestamp` which will automatically return the UTC based Unix-Timestamp.
+
+These are though actions that are not part of the contract which this interface defines.
```
Comment (shadowhand) [inline proposed/clock-meta.md:71]:
```suggestion
```

File: proposed/clock-meta.md [hunk 5]
```diff
@@ -64,25 +82,6 @@ final class SystemClock implements \Psr\Clock\ClockInterface
...
-    public function __construct()
-    {
-        $this->utcTimeZone = new \DateTimeZone('UTC');
-    }
-
-    public function now(): \DateTimeImmutable
-    {
-        return new \DateTimeImmutable('now', $this->utcTimeZone);
-    }
-}
```
Comment (shadowhand) [inline proposed/clock-meta.md:82]: Why was this removed?

File: proposed/clock-meta.md [hunk 6]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
   described in this document, so it is not a coding standard;
 * This PSR does not provide a recommendation on how to handle timezones when 
   retrieving the current time. This is left up to the implementation.
+* This PSR does not handle any Scheduler specific tasks like `sleep` or `wait` as those – even though they take a time or an interval as parameter – are not related to retrieving the current time. Handling such tasks is left to the respective implementations or a different PSR.
```
Comment (shadowhand) [inline proposed/clock-meta.md:42]:
```suggestion
* This PSR does not handle any scheduling methods like `sleep()` or `wait()` because such methods are not related to retrieving the current time.
```

File: proposed/clock-meta.md [hunk 7]
```diff
@@ -2,21 +2,19 @@
...
-`Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time & date 
-functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It does
-not fully mock time when called from other libraries that rely on the system time. 
-`Cake\Chronos\Chronos` does provide mocking, however it is set via a global (static class 
-property), and this has its own pitfalls as it provides no isolation.
+There are currently a few libraries that provide this functionality, however there is no interopability between these different libraries, as they ship with their own clock interfaces. 
+
+Symfony provides a package called `symfony/phpunit-bridge` that has a
+`Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time & date functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It does not fully mock time when called from other libraries that rely on the system time. 
+`Cake\Chronos\Chronos` does provide mocking, however it is set via a global (static class property), and this has its own pitfalls as it provides no isolation.
```
Comment (shadowhand) [inline proposed/clock-meta.md:17]:
```suggestion
Symfony provides a package called `symfony/phpunit-bridge` that has a `Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time and date functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It also does not fully mock time when called from other libraries that rely on the system time.

`Cake\Chronos\Chronos` does provide mocking via a static `setTestNow()` method, but this has its own pitfalls as it provides no isolation and must be called again to stop mocking.
```

**Review by shadowhand — CHANGES_REQUESTED** [review]

File: proposed/clock-meta.md (hunk 5 above)
Comment (heiglandreas) [reply proposed/clock-meta.md:82]: If we are proposing that the timezone is not part of the Interface we should not provide a minimum implementation that provides a timezone specific implementation. And for a minimum implementation the UTCClock is not necessary. If people want a more sophisticated example they can check out one of the multiple provided implementations

File: proposed/clock-meta.md [hunk 8]
```diff
@@ -2,21 +2,19 @@
 
 ## 1. Summary
 
-The purpose of using the `ClockInterface` is to provide a standard way to access the system 
-time, that would allow interopability when testing code that relies on the current time 
-rather than relying on installing PHP extensions or use hacks like re-declaring the `time()`
-function in other namespaces. 
+Getting the current time in applications is a rather straight forward thing given the languages native `time()` function (or one of it's similar functionlaities like `utime()` or `new DateTimeImmutable()` etc.)
```
Comment (benedmunds) [inline proposed/clock-meta.md:5]:
```suggestion
Getting the current time in applications is a rather straight forward thing given the languages native `time()` function (or one of it's similar functionalities like `utime()` or `new DateTimeImmutable()` etc.)
```

File: proposed/clock-meta.md [hunk 9]
```diff
@@ -2,21 +2,19 @@
...
+
+Due to the nature of time as constantly progressing it becomes a nuisance to use these functions if you need predictable results like for testing.
```
Comment (benedmunds) [inline proposed/clock-meta.md:7]:
```suggestion
Due to the nature of time as constantly progressing it becomes a nuisance to use these functions if you need predictable results, for example when testing.
```

File: proposed/clock-meta.md [hunk 10]
```diff
@@ -2,21 +2,19 @@
...
+
+This `ClockInterface` aims to provide a standard way to consume time, that allows interopability not only when consuming the "real" time but also when predictable results need to be available. This allows to skip using PHP Extensions for testing or using hacks like re-declaring the `time()` function in other namespaces.
```
Comment (benedmunds) [inline proposed/clock-meta.md:9]:
```suggestion
This `ClockInterface` aims to provide a standard way to consume time, that allows interoperability not only when consuming the "real" time but also when predictable results need to be available. This allows the avoidance of using PHP Extensions for testing or using hacks like re-declaring the `time()` function in other namespaces. 
```

File: proposed/clock-meta.md [hunk 11]
```diff
@@ -41,17 +39,37 @@ calling `time()` or `date()`.
...
+We have decided to formalize the existing practices, used by several other packages out in the wild. Some of the popular packages providing this functionality are: 
+
+* [`lcobucci/clock`](https://packagist.org/packages/lcobucci/clock)
+* [`kreait/clock`](https://packagist.org/packages/kreait/clock)
+* [`ergebnis/clock`](https://packagist.org/packages/ergebnis/clock)
+* [`mangoweb/clock`](https://packagist.org/packages/mangoweb/clock)
+
+(This list is not exhaustive!)
+
+Some providing
```
Comment (benedmunds) [inline proposed/clock-meta.md:57]:
```suggestion
```

**Review by benedmunds — CHANGES_REQUESTED** [review]

File: proposed/clock-meta.md (hunk 8 above)
Comment (ashnazg) [inline proposed/clock-meta.md:5]: s/straight forward/straightforward/ (it's one word)

s/given the languages native/given the language's native/ (make it possessive)

s/one of it's similar/one of its similar/ (make it possessive rather than "it is" contraction)

Comment (heiglandreas) [reply proposed/clock-meta.md]: Fixed with f5c2188

File: proposed/clock-meta.md (hunk 4 above)
Comment (heiglandreas) [reply proposed/clock-meta.md:70]: We should make it explicit that handling the timezones are not part of the interfaces contract!

**Comment by cseufert** [comment]
Looks great thanks @heiglandreas

**Comment by heiglandreas** [comment]
Anything I can do  to go on with this PSR?

File: proposed/clock-meta.md [hunk 12]
```diff
@@ -2,21 +2,19 @@
 
 ## 1. Summary
 
-The purpose of using the `ClockInterface` is to provide a standard way to access the system 
-time, that would allow interopability when testing code that relies on the current time 
-rather than relying on installing PHP extensions or use hacks like re-declaring the `time()`
-function in other namespaces. 
+Getting the current time in applications is a rather straightforward thing given the language's native `time()` function (or one of its similar functionalities like `utime()` or `new DateTimeImmutable()` etc.)
```
Comment (nicolas-grekas) [inline proposed/clock-meta.md:5]: s/utime/microtime I suppose?

Comment (heiglandreas) [reply proposed/clock-meta.md:5]: Good catch! Thanks. Mixed up the underlying C-functions there 😁

File: proposed/clock-meta.md [hunk 13]
```diff
@@ -2,21 +2,19 @@
...
-`Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time & date 
-functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It does
-not fully mock time when called from other libraries that rely on the system time. 
-`Cake\Chronos\Chronos` does provide mocking, however it is set via a global (static class 
-property), and this has its own pitfalls as it provides no isolation.
+There are currently a few libraries that provide this functionality, however there is no interopability between these different libraries, as they ship with their own clock interfaces. 
+
+Symfony provides a package called `symfony/phpunit-bridge` that has a `Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time and date functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It also does not fully mock time when called from other libraries that rely on the system time.
+
+`Cake\Chronos\Chronos` does provide mocking via a static `setTestNow()` method, but this has its own pitfalls as it provides no isolation and must be called again to stop mocking.
```
Comment (kylekatarnls) [inline proposed/clock-meta.md:17]: `Chronos` just inherits `setTestNow()` from `Carbon`, the project they forked, which implemented it years before. By respect for @briannesbitt, the original author, please mention `Carbon` instead.

File: proposed/clock-meta.md [hunk 14]
```diff
@@ -41,17 +39,36 @@ calling `time()` or `date()`.
...
+We have decided to formalize the existing practices, used by several other packages out in the wild. Some of the popular packages providing this functionality are: 
+
+* [`lcobucci/clock`](https://packagist.org/packages/lcobucci/clock)
+* [`kreait/clock`](https://packagist.org/packages/kreait/clock)
+* [`ergebnis/clock`](https://packagist.org/packages/ergebnis/clock)
+* [`mangoweb/clock`](https://packagist.org/packages/mangoweb/clock)
+
+(This list is not exhaustive!)
+
 interfaces, and some relying on overloading (extending) the Clock class to mock the
```
Comment (kylekatarnls) [inline proposed/clock-meta.md:57]: "Some providing" got truncated.
```suggestion
Some providing interfaces, and some relying on overloading (extending) the Clock class to mock the
```

File: proposed/clock-meta.md [hunk 15]
```diff
@@ -41,17 +39,36 @@ calling `time()` or `date()`.
...
+
+### 4.2 Timezones
+
+Time by now is defined by interaction of electromagnetic radiation with the excited states of certain atoms where the SI defines one second as the duration of 9192631770 cycles of radiation corresponding to the transition between two energy levels of the ground state of the caesium-133 atom at 0K. This means that retrieving the current time will always return the same time, no matter where it is observed. While the timezone defines *where* the time was observed it does not modify the actual "slice" of time.
+
+This means that for the sake of this PSR the timezone is considered an implementation detail of the interface. 
+
+It is up to the implementation to make sure that the timezone is handled according to the business logic of the application. That is either by making sure that a call to `now()` will only return a `DateTimeImmutable` object with a known timezone (implicit contract) or by explicitly changing the timezone to be correct for the application. This an be done by calling `setTimezone()` to create a new `DateTimeImmutable` object with the given timezone.
```
Comment (kylekatarnls) [inline proposed/clock-meta.md:68]: Yes x 1000! This paragraph is the most important to me. I hope the one above about caesium-133 atom won't loose readers before they reach this 🤣

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
Few typos sound to have truncated word/sentence

File: proposed/clock-meta.md (hunk 13 above)
Comment (heiglandreas) [reply proposed/clock-meta.md]: This should now be fixed by mentioning Carboin as well as Chronos

File: proposed/clock-meta.md (hunk 14 above)
Comment (heiglandreas) [reply proposed/clock-meta.md]: Thanks! Fixed that

File: proposed/clock-meta.md (hunk 15 above)
Comment (heiglandreas) [reply proposed/clock-meta.md]: Good catch! Thanks for spotting!

File: proposed/clock-meta.md [hunk 16]
```diff
@@ -41,17 +39,36 @@ calling `time()` or `date()`.
...
+We have decided to formalize the existing practices, used by several other packages out in the wild. Some of the popular packages providing this functionality are: 
+
+* [`lcobucci/clock`](https://packagist.org/packages/lcobucci/clock)
+* [`kreait/clock`](https://packagist.org/packages/kreait/clock)
+* [`ergebnis/clock`](https://packagist.org/packages/ergebnis/clock)
+* [`mangoweb/clock`](https://packagist.org/packages/mangoweb/clock)
+
+(This list is not exhaustive!)
+
+Some of these providing interfaces, and some relying on overloading (extending) the Clock class to mock the
```
Comment (shadowhand) [inline proposed/clock-meta.md:57]:
```suggestion
Some of these provide interfaces and some rely on extending a clock class to mock the
```

File: proposed/clock-meta.md [hunk 17]
```diff
@@ -2,21 +2,19 @@
 
 ## 1. Summary
 
-The purpose of using the `ClockInterface` is to provide a standard way to access the system 
-time, that would allow interopability when testing code that relies on the current time 
-rather than relying on installing PHP extensions or use hacks like re-declaring the `time()`
-function in other namespaces. 
+Getting the current time in applications is a rather straightforward thing given the language's native `time()` function (or one of its similar functionalities like `microtime()` or `new DateTimeImmutable()` etc.)
```
Comment (shadowhand) [inline proposed/clock-meta.md:5]:
```suggestion
Getting the current time in applications is typically achieved using the `time()` or `microtime` functions, or by using a `new \DateTimeImmutable()` class.
```

File: proposed/clock-meta.md [hunk 18]
```diff
@@ -2,21 +2,19 @@
...
+
+Due to the nature of time as constantly progressing it becomes a nuisance to use these functions if you need predictable results, for example when testing.
```
Comment (shadowhand) [inline proposed/clock-meta.md:7]:
```suggestion
Due to the nature of time progression these methods cannot be used when predictable results are needed, such as during testing.
```

**Review by shadowhand — CHANGES_REQUESTED** [review]

File: proposed/clock-meta.md [hunk 19]
```diff
@@ -2,21 +2,19 @@
...
+
+This `ClockInterface` aims to provide a standard way to consume time, that allows interoperability not only when consuming the "real" time but also when predictable results need to be available. This allows the avoidance of using PHP Extensions for testing or using hacks like re-declaring the `time()` function in other namespaces.
```
[the remark this answers is not in the document]
Comment (heiglandreas) [reply proposed/clock-meta.md:9]: In that context rewriting a function in a namespace *is* a hack ;-)

But I do get your point :wink:

File: proposed/clock-meta.md [hunk 20]
```diff
@@ -41,17 +39,36 @@ calling `time()` or `date()`.
...
-interfaces, and some relying on overloading (extending) the Clock class to mock the
+We have decided to formalize the existing practices, used by several other packages out in the wild. Some of the popular packages providing this functionality are: 
+
+* [`lcobucci/clock`](https://packagist.org/packages/lcobucci/clock)
+* [`kreait/clock`](https://packagist.org/packages/kreait/clock)
+* [`ergebnis/clock`](https://packagist.org/packages/ergebnis/clock)
+* [`mangoweb/clock`](https://packagist.org/packages/mangoweb/clock)
+
+(This list is not exhaustive!)
+
```
[the remark this answers is not in the document]
Comment (heiglandreas) [reply proposed/clock-meta.md:56]: Even though that should not be necessary I'd rather keep it to make clear that there might be other packages that provide the same value. Yes, we introduced that list with "some popular packages … are". But there will be people not reading that. And the oneliner right underneath the table is catching the eye better than the message in the copy-text above.

File: proposed/clock-meta.md [hunk 21]
```diff
@@ -41,17 +39,36 @@ calling `time()` or `date()`.
...
+### 4.2 Timezones
+
+Time by now is defined by interaction of electromagnetic radiation with the excited states of certain atoms where the SI defines one second as the duration of 9192631770 cycles of radiation corresponding to the transition between two energy levels of the ground state of the caesium-133 atom at 0K. This means that retrieving the current time will always return the same time, no matter where it is observed. While the timezone defines *where* the time was observed it does not modify the actual "slice" of time.
+
+This means that for the sake of this PSR the timezone is considered an implementation detail of the interface. 
+
+It is up to the implementation to make sure that the timezone is handled according to the business logic of the application. That is either by making sure that a call to `now()` will only return a `DateTimeImmutable` object with a known timezone (implicit contract) or by explicitly changing the timezone to be correct for the application. This can be done by calling `setTimezone()` to create a new `DateTimeImmutable` object with the given timezone. 
+
+These are though actions that are not part of the contract which this interface defines.
```
Comment (shadowhand) [inline proposed/clock-meta.md:70]:
```suggestion
These actions are not defined in this interface.
```

**Review by shadowhand — APPROVED** [review]
One remaining suggestion, but otherwise looks great!

**Review by benedmunds — APPROVED** [review]

**Review by lcobucci — APPROVED** [review]

**Review by cseufert — APPROVED** [review]
